### PR TITLE
workflow: remove promise in workflow

### PR DESCRIPTION
**Pull request by masco-enovance**

### CODE REVIEW

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb
run skydive-cdd-overview-tests

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by masco-enovance** [comment]
run skydive-cdd-overview-tests

**Comment by masco-enovance** [comment]
run skydive-cdd-overview-tests

**Comment by masco-enovance** [comment]
rerun skydive-cdd-overview-tests

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

File: js/api.ts [hunk 1]
```diff
@@ -70,22 +70,18 @@ if (jsEnv) {
...
-            try {
-                var output = request(url, method, body)
-                if (output && output.length > 0) {
-                    data = JSON.parse(output)
-                } else {
-                    data = null
-                }
-                resolve(data)
+        var data: any
+        try {
```
Comment (safchain) [inline js/api.ts]: nit: there is no need to try/catch if we just throw e on error

**Comment by hunchback** [comment]
I don't know if related to your changes (or an issue on master), but I got the following race condition on the analyzer:

```
==================
WARNING: DATA RACE
Write at 0x00c4215a62e0 by goroutine 270:
  github.com/skydive-project/skydive/graffiti/graph.(*Graph).NodeUpdated()
      /home/vagrant/src/github.com/skydive-project/skydive/graffiti/graph/graph.go:644 +0xb0
  github.com/skydive-project/skydive/graffiti/hub.(*TopologyAgentEndpoint).OnStructMessage()
      /home/vagrant/src/github.com/skydive-project/skydive/graffiti/hub/pod_endpoint.go:104 +0x2138
  github.com/skydive-project/skydive/websocket.(*structSpeakerEventDispatcher).dispatchMessage()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/message.go:268 +0x21b
  github.com/skydive-project/skydive/websocket.(*StructSpeaker).OnMessage()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/message.go:387 +0x3f6
  github.com/skydive-project/skydive/websocket.(*Conn).run.func2()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/client.go:320 +0x1e1
  github.com/skydive-project/skydive/websocket.(*Conn).run()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/client.go:394 +0x3cc

Previous read at 0x00c4215a62e0 by goroutine 279:
  [failed to restore the stack]

Goroutine 270 (running) created at:
  github.com/skydive-project/skydive/websocket.(*Conn).Start()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/client.go:300 +0x6b
  github.com/skydive-project/skydive/websocket.(*Server).newIncomingClient()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/server.go:166 +0x1033
  github.com/skydive-project/skydive/websocket.NewStructServer.func1()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/message.go:509 +0xe4
  github.com/skydive-project/skydive/websocket.(*Server).serveMessages()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/server.go:104 +0xf62
  github.com/skydive-project/skydive/websocket.(*Server).(github.com/skydive-project/skydive/websocket.serveMessages)-fm()
      /home/vagrant/src/github.com/skydive-project/skydive/websocket/server.go:183 +0x64
  github.com/skydive-project/skydive/http.(*Server).HandleFunc.func1()
      /home/vagrant/src/github.com/skydive-project/skydive/http/server.go:165 +0x223
  github.com/skydive-project/skydive/http.(*NoAuthenticationBackend).Wrap.func1()
      /home/vagrant/src/github.com/skydive-project/skydive/http/noauth.go:56 +0x154
  github.com/skydive-project/skydive/http.(*Server).HandleFunc.func2()
      /home/vagrant/src/github.com/skydive-project/skydive/http/server.go:174 +0x8a
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1918 +0x51
  github.com/skydive-project/skydive/vendor/github.com/gorilla/mux.(*Router).ServeHTTP()
      /home/vagrant/src/github.com/skydive-project/skydive/vendor/github.com/gorilla/mux/mux.go:150 +0x265
  github.com/skydive-project/skydive/vendor/github.com/gorilla/handlers.CompressHandlerLevel.func1()
      /home/vagrant/src/github.com/skydive-project/skydive/vendor/github.com/gorilla/handlers/compress.go:146 +0x754
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1918 +0x51
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2619 +0xbc
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1801 +0x83b

Goroutine 279 (running) created at:
  net/http.(*Server).Serve()
      /usr/local/go/src/net/http/server.go:2720 +0x37c
  github.com/skydive-project/skydive/http.(*Server).Serve()
      /home/vagrant/src/github.com/skydive-project/skydive/http/server.go:128 +0x59f
  github.com/skydive-project/skydive/analyzer.(*Server).Start.func1()
      /home/vagrant/src/github.com/skydive-project/skydive/analyzer/server.go:138 +0x7d
==================
```

File: js/api.ts [hunk 2]
```diff
@@ -193,6 +189,7 @@ export class API<T extends APIObject> {
         let resource = this.Resource;
         return this.client.request("/api/" + resource, "POST", JSON.stringify(obj), {})
             .then(function (data) {
+                console.log(data)
```
Comment (hunchback) [inline js/api.ts]: I am not sure we want to leave this `console.log()`

File: js/api.ts [hunk 3]
```diff
@@ -939,5 +929,5 @@ export class Client {
 }
 
 export function sleep(time) {
-    return new Promise(function(resolve) { return setTimeout(resolve, time); })
+  return new Promise(function(resolve) { return setTimeout(resolve, time); })
```
Comment (hunchback) [inline js/api.ts]: indentation should be 4 not 2

File: statics/workflows/flow-matrix.yaml [hunk 4]
```diff
@@ -345,26 +345,14 @@ Source: |
...
-                return outputResult()
-            })
+        var nodes = G.V().Has('Type', 'host').HasKey('Sockets').result()
+        var interfaces = G.V().HasKey('IPV4').result()
+        buildCaches(nodes, interfaces)
+        topologyMatrix(nodes)
+        if (useFlows) {
+            var sockets = G.Flows().Has('Transport', NE('')).Sockets().result()
+            flowMatrix(sockets)
+            return outputResult()
```
Comment (hunchback) [inline statics/workflows/flow-matrix.yaml]: this line is redundant and can be removed

**Comment by hunchback** [comment]
@masco-enovance - indentation is in-consistent in TypeScript files - it should be 2 but is sometimes 4 (for example in places you removed surrounding try ... catch clauses.

File: js/api.ts (hunk 3 above)
Comment (hunchback) [inline js/api.ts]: Do we not want to use await? Such as:

```
export function sleep(ms) {
  await new Promise(function(resolve) {
    setTimeout(resolve, ms)
  })
}
```

File: statics/workflows/flow-matrix.yaml (hunk 4 above)
Comment (masco-enovance) [reply statics/workflows/flow-matrix.yaml]: done!

File: js/api.ts (hunk 3 above)
Comment (masco-enovance) [reply js/api.ts]: currently, await and async are not supported in otto engine.

File: js/api.ts (hunk 2 above)
Comment (masco-enovance) [reply js/api.ts]: yep, added for debugging purpose and forgot to remove.

File: js/api.ts (hunk 1 above)
Comment (masco-enovance) [reply js/api.ts]: removed

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb

File: js/api.ts (hunk 3 above)
Comment (safchain) [reply js/api.ts]: @masco-enovance moreover I think this function should be removed as not used anymore ?

Comment (masco-enovance) [reply js/api.ts]: replaced with otto's sleep function

File: statics/workflows/check-mtu.yaml [hunk 5]
```diff
@@ -12,18 +12,16 @@ Parameters:
...
-                    if (MTU != undefined && (node.Metadata === undefined || node.Metadata.MTU < MTU)) {
-                        console.log("MTU " + node.Metadata.MTU + " on node " + node.ID + " is inferior to " + MTU)
-                        return false
-                    }
-                    if (node.Metadata !== undefined)
-                        MTU = node.Metadata.MTU
-                }
-                return true
-            })
+        var path = client.gremlin.G().V().Has('TID', from).ShortestPathTo(Metadata('TID', to)).result()
```
Comment (dvandra) [inline statics/workflows/check-mtu.yaml]: Maybe the path is not exactly the shortest-path.

File: statics/workflows/check-mtu.yaml [hunk 6]
```diff
@@ -12,18 +12,16 @@ Parameters:
...
+        for (var i in path) {
+          var node = path[i]
+          console.log(node)
+          if (MTU != undefined && (node.Metadata === undefined || node.Metadata.MTU < MTU)) {
+             console.log("MTU " + node.Metadata.MTU + " on node " + node.ID + " is inferior to " + MTU)
+             return false
+          }
+          if (node.Metadata !== undefined)
+             MTU = node.Metadata.MTU
```
Comment (dvandra) [inline statics/workflows/check-mtu.yaml]: Can we show the correct MTU in result?

**Pull request merged**